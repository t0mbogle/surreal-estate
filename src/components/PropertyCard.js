import React from "react";
import PropTypes from "prop-types";
import {
  FaFortAwesome,
  FaBed,
  FaBath,
  FaPoundSign,
  FaEnvelope,
} from "react-icons/fa";
import "../styles/property-card.css";

const PropertyCard = ({
  title,
  type,
  city,
  bathrooms,
  bedrooms,
  price,
  email,
}) => {
  // const formattedPrice = parseInt(price, 10).toLocaleString("en-GB", {
  //   minimumFractionDigits: 2,
  //   maximumFractionDigits: 2,
  // });

  const formatPrice = Intl.NumberFormat("en-GB", {
    style: "currency",
    currency: "GBP",
  });
  // eslint-disable-next-line no-var
  const newPrice = formatPrice.format(price);

  return (
    <>
      <div className="property-card">
        <b>Property Listing</b> <FaFortAwesome />
      </div>
      <div className="property-card_title">{title}</div>
      <div className="property-card_typeCity">
        {type} - {city}
      </div>
      <div className="property-card_bathroom">
        <FaBath />
        {bathrooms}
      </div>
      <div className="property-card_bedroom">
        <FaBed />
        {bedrooms}
      </div>
      <div className="property-card_price">
        <FaPoundSign />
        {newPrice}
      </div>
      <div className="property-card_email">
        <a href={`mailto:${email}`}>
          <FaEnvelope /> Contact
        </a>
      </div>
    </>
  );
};

export default PropertyCard;

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  email: PropTypes.string.isRequired,
};
