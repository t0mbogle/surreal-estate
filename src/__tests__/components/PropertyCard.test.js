import React from "react";
import { render, screen } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

describe("Property Card", () => {
  const validProps = {
    title: "Two bed flat",
    type: "Flat",
    city: "Manchester",
    bathrooms: 1,
    bedrooms: 2,
    price: 750,
    email: "hello@email.com",
  };

  it("renders property card with correct data", () => {
    const { asFragment } = render(<PropertyCard />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("checks title prop has rendered correctly", () => {
    const { getByText } = render(<PropertyCard {...validProps} />);

    expect(getByText(/two bed flat/i).textContent).toBe("Two bed flat");
  });
  it("checks type and city props have rendered correctly", () => {
    const { getByText } = render(<PropertyCard {...validProps} />);

    expect(getByText(/flat - manchester/i).textContent).toBe(
      "Flat - Manchester"
    );
  });
  it("checks bathrooms prop has rendered correctly", () => {
    const { getByText } = render(<PropertyCard {...validProps} />);

    expect(getByText(/1/).textContent).toBe("1");
  });
  it("checks bedrooms prop has rendered correctly", () => {
    const { getByText } = render(<PropertyCard {...validProps} />);

    expect(getByText(/2/).textContent).toBe("2");
  });
  it("checks price prop has rendered correctly", () => {
    const { getByText } = render(<PropertyCard {...validProps} />);

    expect(getByText(/750/).textContent).toBe("750");
  });
  it("checks email prop has rendered correctly", () => {
    // const { getByText } = render(<PropertyCard {...validProps} />);
    const linkEl = screen.getByRole("button", { name: "email" });

    expect(linkEl).toHaveAttribute("href", "email");
    // expect(getByText("a").getAttribute("href")).toBeInTheDocument();
    // expect(getByText("email").closest("a")).toHaveAttribute("href");
    // expect(screen.getByRole("a")).toContain("email");
    // expect(getByRole(/email/i).textContent).toContain("email");
  });
});
