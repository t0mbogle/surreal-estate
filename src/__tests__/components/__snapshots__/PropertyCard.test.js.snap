// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Card renders property card with correct data 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <b>
      Property Listing
    </b>
     
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M489.2 287.9h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6V146.2c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-6-8-4.6-11.7-4.6v-38c8.3-2 17.1-3.4 25.7-3.4 10.9 0 20.9 4.3 31.4 4.3 4.6 0 27.7-1.1 27.7-8v-60c0-2.6-2-4.6-4.6-4.6-5.1 0-15.1 4.3-24 4.3-9.7 0-20.9-4.3-32.6-4.3-8 0-16 1.1-23.7 2.9v-4.9c5.4-2.6 9.1-8.3 9.1-14.3 0-20.7-31.4-20.8-31.4 0 0 6 3.7 11.7 9.1 14.3v111.7c-3.7 0-11.7-1.4-11.7 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32H128v-32c0-2.6-2-4.6-4.6-4.6H96c-2.6 0-4.6 2-4.6 4.6v178.3H54.8v-32c0-2.6-2-4.6-4.6-4.6H22.8c-2.6 0-4.6 2-4.6 4.6V512h182.9v-96c0-72.6 109.7-72.6 109.7 0v96h182.9V292.5c.1-2.6-1.9-4.6-4.5-4.6zm-288.1-4.5c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64zm146.4 0c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64z"
      />
    </svg>
  </div>
  <div
    class="property-card_title"
  />
  <div
    class="property-card_typeCity"
  >
     - 
  </div>
  <div
    class="property-card_bathroom"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
      />
    </svg>
  </div>
  <div
    class="property-card_bedroom"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 640 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
      />
    </svg>
  </div>
  <div
    class="property-card_price"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M308 352h-45.495c-6.627 0-12 5.373-12 12v50.848H128V288h84c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-84v-63.556c0-32.266 24.562-57.086 61.792-57.086 23.658 0 45.878 11.505 57.652 18.849 5.151 3.213 11.888 2.051 15.688-2.685l28.493-35.513c4.233-5.276 3.279-13.005-2.119-17.081C273.124 54.56 236.576 32 187.931 32 106.026 32 48 84.742 48 157.961V224H20c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h28v128H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h296c6.627 0 12-5.373 12-12V364c0-6.627-5.373-12-12-12z"
      />
    </svg>
    Â£NaN
  </div>
  <div
    class="property-card_email"
  >
    <a
      href="mailto:undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
        />
      </svg>
       Contact
    </a>
  </div>
</DocumentFragment>
`;
